File: backend/src/main/java/com/chronovault/entity/Alert.java
package com.chronovault.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "alerts")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Alert {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "server_id")
    private Server server;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private AlertSeverity severity;

    @Column(nullable = false, length = 300)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 100)
    private String source;

    @Column(length = 50)
    private String category;

    @Column(name = "root_cause_analysis", columnDefinition = "TEXT")
    private String rootCauseAnalysis;

    @Column(name = "storage_percent")
    private Integer storagePercent;

    @Column(name = "growth_rate", length = 50)
    private String growthRate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private AlertStatus status;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (status == null) status = AlertStatus.OPEN;
    }

    public boolean hasAutoFix() {
        return category != null && ("磁盘".equals(category) || "存储".equals(category) || "容器".equals(category));
    }

    public enum AlertSeverity { CRITICAL, PREDICTIVE, WARNING }
    public enum AlertStatus { OPEN, RESOLVED, DISMISSED }
}

File: frontend/src/types/alert.ts
import type { Severity } from './common'

export interface Alert {
  id: number
  title: string
  description?: string
  message?: string
  severity: Severity
  status: string
  source?: string
  aiAnalysis?: string
  time?: string
  createdAt: string
  updatedAt?: string
}

export interface AlertRule {
  id: number
  name: string
  metric: string
  threshold: number
  durationMinutes: number
  severity: string
  enabled?: boolean
}

export interface CreateAlertRuleRequest {
  name: string
  metric: string
  threshold: number
  durationMinutes: number
  severity: string
}

export interface AlertStats {
  total: number
  critical: number
  warning: number
  info: number
  predictive?: number
  resolvedCount?: number
}
